### ISSUE TITLE
Issue 4450: author is not detected in .h files

### ISSUE THREAD

**Issue by abraemer** [mstykow/scancode-rust#264]
Author detection fails for certain author name formats in header file comments.

## Key Details

- The `@author` tag in header comments is not being detected for certain name formats
- Example that fails detection:

  ```c
  /*!
   * @brief       This api Initializes diag state machine
   * @author      HSamaa
   */
  ```

- Workaround: Separating "HSamaa" to "H. Samaa" makes detection work
- ScanCode version: 32.2.1
- The issue is with the author name parsing/detection logic

## DIFFICULTY: EASY

Clear reproduction case with a workaround identified. The issue is in the author name pattern matching logic.

## Upstream Reference
- Issue: aboutcode-org/scancode-toolkit#4450
- URL: https://github.com/aboutcode-org/scancode-toolkit/issues/4450
- Title: author is not detected in .h files
- State: open

## Upstream Context
in the following header the Author is not detected. If i separate it to H.

**Comment by mstykow** [comment]
Final clarification: this issue is non-reproducible in current scancode-rust. The HSamaa author pattern was validated with a dedicated fixture in the authors golden dataset, and the golden authors test passed. Closing is intentional because current behavior already detects this case correctly.
